ipa-core/src/helpers/mod.rs

```diff
@@ -437,13 +438,16 @@ impl<V: SharedValue> Message for V {}
 
 impl Serializable for PublicKey {
     type Size = typenum::U32;
+    type DeserializationError = Infallible;
 
     fn serialize(&self, buf: &mut GenericArray<u8, Self::Size>) {
         buf.copy_from_slice(self.as_bytes());
     }
 
-    fn deserialize(buf: &GenericArray<u8, Self::Size>) -> Self {
-        Self::from(<[u8; 32]>::from(*buf))
+    fn deserialize(
+        buf: &GenericArray<u8, Self::Size>,
+    ) -> std::result::Result<Self, Self::DeserializationError> {
+        Ok(Self::from(<[u8; 32]>::from(*buf)))
```

Comment:
This may also be a place to check for a canonical encoding.

Reply:
I may misunderstand something, but I thought 25519 accepts any 32 byte value as a valid [public key](https://en.wikipedia.org/wiki/Curve25519)